src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateGeneralLetter.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.divorce.caseworker.service.task;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.type.Document;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.GeneralLetter;
+import uk.gov.hmcts.divorce.divorcecase.model.GeneralLetterDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.GeneralParties;
+import uk.gov.hmcts.divorce.divorcecase.model.LanguagePreference;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+import uk.gov.hmcts.divorce.document.CaseDataDocumentService;
+import uk.gov.hmcts.divorce.document.DocumentIdProvider;
+import uk.gov.hmcts.divorce.document.content.GeneralLetterTemplateContent;
+
+import java.time.Clock;
+import java.util.Collection;
+import java.util.List;
+
+import static java.time.LocalDateTime.now;
+import static java.util.stream.Stream.ofNullable;
+import static uk.gov.hmcts.divorce.caseworker.service.task.util.FileNameUtil.formatDocumentName;
+import static uk.gov.hmcts.divorce.divorcecase.model.CaseDocuments.addDocumentToTop;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.GENERAL_LETTER_DOCUMENT_NAME;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.GENERAL_LETTER_TEMPLATE_ID;
+
+@Component
+@Slf4j
+public class GenerateGeneralLetter implements CaseTask {
+
+    @Autowired
+    private CaseDataDocumentService caseDataDocumentService;
+
+    @Autowired
+    private GeneralLetterTemplateContent templateContent;
+
+    @Autowired
+    private Clock clock;
+
+    @Autowired
+    private DocumentIdProvider documentIdProvider;
+
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+
+        final Long caseId = caseDetails.getId();
+        final CaseData caseData = caseDetails.getData();
+
+        log.info("Generating general letter for case id: {} ", caseId);
+
+        LanguagePreference languagePreference =
+            GeneralParties.RESPONDENT.equals(caseData.getGeneralLetter().getGeneralLetterParties())
+            ? caseData.getApplicant2().getLanguagePreference()
+            : caseData.getApplicant1().getLanguagePreference();
+
+        Document generalLetterDocument = caseDataDocumentService.renderDocument(
```

Comment:
you could use  renderDocumentAndUpdateCaseData which generates and add document to documents generated ?